smithy-kotlin-codegen/src/test/kotlin/software/amazon/smithy/kotlin/codegen/rendering/protocol/HttpProtocolClientGeneratorTest.kt

```diff
@@ -145,7 +159,10 @@ class HttpProtocolClientGeneratorTest {
             }
         }
         op.install(MockMiddleware(configurationField1 = "testing"))
-        return op.execute(client, input, block)
+        val rootSpan = config.tracer.createRootSpan("GetFooStreamingOutput-${'$'}{op.context.sdkRequestId}")
+        return coroutineContext.withRootTraceSpan(rootSpan) {
```

Comment:
nit: Still would have loved to see some sugar for `withRootTraceSpan` and `withSpan` 😛 

```kotlin
inline suspend fun<T> withSpan(block: () -> T): T = coroutineContext.withSpan(...)
```

such that call sites can just do:

```
suspend fun foo() = withSpan("foo") {...}
```

This is a super minor style thing at the end of the day